src/backend/benchmark/tpcc/tpcc_loader.cpp

```diff
@@ -0,0 +1,1426 @@
+  customer_columns.push_back(c_state_column);
+  auto c_zip_column = catalog::Column(VALUE_TYPE_VARCHAR, zip_length, "C_ZIP", is_inlined);
+  customer_columns.push_back(c_zip_column);
+  auto c_phone_column = catalog::Column(VALUE_TYPE_VARCHAR, phone_length, "C_PHONE", is_inlined);
+  customer_columns.push_back(c_phone_column);
+  auto c_since_column = catalog::Column(VALUE_TYPE_TIMESTAMP, GetTypeSize(VALUE_TYPE_TIMESTAMP), "C_SINCE", is_inlined);
+  customer_columns.push_back(c_since_column);
+  auto c_credit_column = catalog::Column(VALUE_TYPE_VARCHAR, credit_length, "C_CREDIT", is_inlined);
+  customer_columns.push_back(c_credit_column);
+  auto c_credit_lim_column = catalog::Column(VALUE_TYPE_DOUBLE, GetTypeSize(VALUE_TYPE_DOUBLE), "C_CREDIT_LIM", is_inlined);
+  customer_columns.push_back(c_credit_lim_column);
+  auto c_discount_column = catalog::Column(VALUE_TYPE_DOUBLE, GetTypeSize(VALUE_TYPE_DOUBLE), "C_DISCOUNT", is_inlined);
+  customer_columns.push_back(c_discount_column);
+  auto c_balance_column = catalog::Column(VALUE_TYPE_DOUBLE, GetTypeSize(VALUE_TYPE_DOUBLE), "C_BALANCE", is_inlined);
+  customer_columns.push_back(c_balance_column);
+  auto c_ytd_payment_column = catalog::Column(VALUE_TYPE_DOUBLE, GetTypeSize(VALUE_TYPE_DOUBLE), "C_YTD_PAYMENT", is_inlined);
+  customer_columns.push_back(c_ytd_payment_column);
+  auto c_payment_column = catalog::Column(VALUE_TYPE_INTEGER, GetTypeSize(VALUE_TYPE_INTEGER), "C_PAYMENT_CNT", is_inlined);
+  customer_columns.push_back(c_payment_column);
+  auto c_delivery_column = catalog::Column(VALUE_TYPE_INTEGER, GetTypeSize(VALUE_TYPE_INTEGER), "C_DELIVERY_CNT", is_inlined);
+  customer_columns.push_back(c_delivery_column);
+  auto c_data_column = catalog::Column(VALUE_TYPE_VARCHAR, data_length, "C_DATA", is_inlined);
+  customer_columns.push_back(c_data_column);
+
+  catalog::Schema *table_schema = new catalog::Schema(customer_columns);
+  std::string table_name("CUSTOMER");
+
+  customer_table = storage::TableFactory::GetDataTable(
+      tpcc_database_oid,
+      customer_table_oid,
+      table_schema, table_name,
+      DEFAULT_TUPLES_PER_TILEGROUP,
+      own_schema,
+      adapt_table);
+
+  tpcc_database->AddTable(customer_table);
+
+  auto tuple_schema = customer_table->GetSchema();
+  std::vector<oid_t> key_attrs;
+  catalog::Schema *key_schema = nullptr;
+  index::IndexMetadata* index_metadata = nullptr;
+
+  // Primary index on C_ID, C_D_ID, C_W_ID
+  key_attrs = {0, 1, 2};
+  key_schema = catalog::Schema::CopySchema(tuple_schema, key_attrs);
+  key_schema->SetIndexedColumns(key_attrs);
+
+  index_metadata = new index::IndexMetadata(
+    "customer_pkey", customer_table_pkey_index_oid, INDEX_TYPE_BTREE,
+    INDEX_CONSTRAINT_TYPE_PRIMARY_KEY, tuple_schema, key_schema, true);
+
+  index::Index *pkey_index = index::IndexFactory::GetInstance(index_metadata);
+  customer_table->AddIndex(pkey_index);
+
+
+  
+  // Secondary index on C_W_ID, C_D_ID, C_LAST
+  key_attrs = {1, 2, 5};
```

Comment:
Instead of referring to columns by offsets and by the order in which they are inserted into the customer_columns vector it might be better to nave an enum or a map from column name to index. This make the mapping more explicit and one need not rely on the comments to know which offset if for what column.